mqtt-client/main.py

```diff
@@ -59,18 +240,86 @@ def __init__(self, webinterface, logger):
+        # sensors
+        self._sensor_config = {
+            'temperature': {
+                'enabled':        self._config.get('temperature_status_enabled'),
+                'topic':          self._config.get('temperature_status_topic_format'),
+                'qos':            int(self._config.get('temperature_status_qos')),
+                'retain':         self._config.get('temperature_status_retain'),
+                'poll_frequency': int(self._config.get('temperature_status_poll_frequency'))
+            },
+            'humidity': {
+                'enabled':        self._config.get('humidity_status_enabled'),
+                'topic':          self._config.get('humidity_status_topic_format'),
+                'qos':            int(self._config.get('humidity_status_qos')),
+                'retain':         self._config.get('humidity_status_retain'),
+                'poll_frequency': int(self._config.get('humidity_status_poll_frequency'))
+            },
+            'brightness': {
+                'enabled':        self._config.get('brightness_status_enabled'),
+                'topic':          self._config.get('brightness_status_topic_format'),
+                'qos':            int(self._config.get('brightness_status_qos')),
+                'retain':         self._config.get('brightness_status_retain'),
+                'poll_frequency': int(self._config.get('brightness_status_poll_frequency'))
+            },
+            'power': {
+                'enabled':        self._config.get('power_status_enabled'),
+                'topic':          self._config.get('power_status_topic_format'),
+                'qos':            int(self._config.get('power_status_qos')),
+                'retain':         self._config.get('power_status_retain'),
+                'poll_frequency': int(self._config.get('power_status_poll_frequency'))
+            },
+            'energy': {
+                'enabled':        self._config.get('energy_status_enabled'),
+                'topic':          self._config.get('energy_status_topic_format'),
+                'qos':            int(self._config.get('energy_status_qos')),
+                'retain':         self._config.get('energy_status_retain'),
+                'poll_frequency': int(self._config.get('energy_status_poll_frequency'))
+            }
+        }
+        # output command
+        self._output_command_topic = self._config.get('output_command_topic')
+        # logging topic
+        self._logging_topic = self._config.get('logging_topic')
+        # timezone
+        self._timezone = self._config.get('timezone')
+        self._enabled = self._hostname is not None and self._port is not None
         self.logger('MQTTClient is {0}'.format('enabled' if self._enabled else 'disabled'))
 
     def _load_configuration(self):
-        # Inputs
+        if self._input_enabled:
+            self._load_input_configuration()
```

Comment:
Sometimes, depending on what starts when, it is possible that a plugin can't immediately load certain information. They will fail with e.g. `Failed to load output configurations`. A good change would be to have something like:

```
def _load_configuration(self):
    inputs_loaded = not self._input_enabled
    ...
    should_load = not all([inputs_loaded, ...])
    while should_load:
        if self._input_enabled and not inputs_loaded:
           inputs_loaded = self._load_input_configuration()
        ...
        should_load = not all([inputs_loaded, ...])
        if should_load:
            time.sleep(15)
```

This will make sure that after every config change or after startup it ensures that all configuration is loaded, and will re-try if needed. The `_load_..._configuration` calls should then obviously return `True` or `False` depending on their success.

Reply:
Change made in [this commit](https://github.com/DaGolle/openmotics-plugins/commit/4d460d4ecb939483dcb0a535284ee7d02ee0b34d).